mitodl/mitxonline
# 1676 design new catalog page

# What are the relevant tickets?
https://github.com/mitodl/hq/issues/1676

# Description

1. Updates the `/courseruns` API endpoint to return all course runs.  Previously, this API endpoint performed some filtering for specific functionality which restricted the ability to use the API endpoint for other uses.
2. Renames the unused "CourseTopic" model in the courses app to "Department".  This model is added to the Courses and Programs model.
3. Adds the new Catalog page at `http://<BASE_URL>/catalog`
4. Updates the `/courses` and `/programs` API endpoints to return paginated results only if a `page` query parameter is specified.

# Screenshots
## Desktop views
![Screenshot 2023-08-14 at 13 10 06](https://github.com/mitodl/mitxonline/assets/8311573/72e4441b-2ac3-415c-8697-7ed6e5bdd8b2)
![Screenshot 2023-08-14 at 13 10 10](https://github.com/mitodl/mitxonline/assets/8311573/4efabed9-6a1b-4c2e-885f-243a381a60e3)
![Screenshot 2023-08-14 at 13 10 15](https://github.com/mitodl/mitxonline/assets/8311573/ee760971-c0ac-460f-847b-2894111f6bee)
![Screenshot 2023-08-14 at 13 10 19](https://github.com/mitodl/mitxonline/assets/8311573/c6dc0e31-f8b2-49f2-900a-f35af7ba14b2)

## Tablet views
![Screenshot 2023-08-14 at 15 21 37](https://github.com/mitodl/mitxonline/assets/8311573/a8bdef1b-c994-489c-8712-b2e641646285)

## Mobile views
![Screenshot 2023-08-14 at 14 37 26](https://github.com/mitodl/mitxonline/assets/8311573/09874e34-9b40-4cee-a820-3b71a12b42eb)
![Screenshot 2023-08-14 at 14 37 33](https://github.com/mitodl/mitxonline/assets/8311573/b13925bb-5824-4965-bd36-55d14022bfc4)
![Screenshot 2023-08-14 at 14 39 15](https://github.com/mitodl/mitxonline/assets/8311573/22f682ca-8185-4e2a-abe4-2f9909fdfdc9)


# How can this be tested?

1. Ensure that the Courses tab is populated with courses which adhere to the logic detailed in this comment: https://github.com/mitodl/hq/issues/1676#issuecomment-1655795103
2. Ensure that the Programs tab is populated with programs which have `live = True`.
3. Ensure that animations and visual features of the page match the mock-ups in the relevant tickets.
5. Ensure that the page can appropriately render for desktop, tablet, and mobile viewports.
6. Ensure that the page can appropriately render within different browsers.
7. Ensure that the page can efficiently and appropriately display a large number of courses and programs (more than 15), in order test that not all catalog items are loaded at once.  12 items should load at first, then as you scroll down the page, more items should load.
8. Ensure that the content on the page is accessible to those using screen readers and keyboards.
9. Review the included unit tests and assess their coverage.
10. Ensure that the GET requests to `/api/courses/?page=1` and `/api/programs/?page=1` return paginated results, while GET requests to `/api/courses/` and `/api/programs/` do not return paginated results.
